File: python/caesar_cipher.py


def caesar_cipher(string, shift):
    """
    This function takes a string and returns the caeser cipher of the string
    rotated by the shift amount.

    Args:
        string (str): The string to be encrypted.
        shift (int): The amount to shift the string.

    Returns:
        str: The encrypted string.

    Example:
        >>> caesar_cipher("Hello, World!", 3)
        'Khoor, Zruog!'
    """

    result = list()
    lower_ascii_offset, upper_ascii_offset = 97, 65

    for char in string:
        if char.isupper():
            result.append(chr((ord(char) - upper_ascii_offset + shift) % 26 + upper_ascii_offset))
        elif char.islower():
            result.append(chr((ord(char) - lower_ascii_offset + shift) % 26 + lower_ascii_offset))
        else:
            result.append(char)

    return "".join(result)


def main():
    print(caesar_cipher("Hello, World!", 3))


if __name__ == '__main__':
    main()

File: js/caesarCipher.js
/**
 * This function will encrypt the given string using the Caesar Cipher.
 *
 * @param {String} string The string to be encrypted
 * @param {Integer} shift The number of places to shift the string
 * @returns {String} The encrypted string
 */
const caesarCipher = (string, shift) => {
  const result = [];

  const lowerAsciiOffset = 97;
  const upperAsciiOffset = 65;

  for (const char of string) {
    if (char.toUpperCase() === char) {
      result.push(
        String.fromCharCode(
          ((char.charCodeAt(0) - upperAsciiOffset + shift) % 26) +
            upperAsciiOffset
        )
      );
    } else if (char.toLowerCase() === char) {
      result.push(
        String.fromCharCode(
          ((char.charCodeAt(0) - lowerAsciiOffset + shift) % 26) +
            lowerAsciiOffset
        )
      );
    } else {
      result.push(char);
    }
  }

  return result.join("");
};

module.exports = { caesarCipher };
